[Android] Revert main layout background transparency to black

quietvoid opened the pull request:
## Description
This improves playback for devices that don't properly support non-fullscreen renders, in Dolby Vision mode only.
Since Kodi renders to a cropped view (using margins), the default background is transparent.
Some devices have problems with this transparent background, and instead renders it as a higher black floor for some reason.
This makes cropped videos (or any that don't fill the view) have noticeable elevated black in letterboxing areas.
For some reason, this only affects Dolby Vision mode.

Known devices with this bug: FireTV Stick 4K Max, FireTV Cube 3rd gen.

The change is now simply a revert of 6f984677ed840f391d78307955a7df39f3016eb8, as discussed.

## Motivation and context
Improves playback quality of Dolby Vision videos that were cropped, and that don't have 16:9 stored dimensions.

## How has this been tested?
Runtime tested on FireTV Stick 4K Max.

## What is the effect on users?
Impacts main app code to remove a workaround added in https://github.com/xbmc/xbmc/pull/15225.

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [x] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

joseluismarti commented:
The call to the native method `setBackgroundColor()` instead of including it in the CJNIMainActivity should be added in the libandroidjni library, it would be `CJNIView::setBackgroundColor()`

quietvoid commented:
You're right, at first I was trying to use mDecorView in the main activity and totally forgot about this.

quietvoid commented:
@joseluismarti Yeah, I can't make it work through a `CJNIView`.
For example, with this patch to `libandroidjni`: https://github.com/quietvoid/libandroidjni/commit/40cde0197f464d66a009a3dc5aa76bd51929c0a8

And Kodi side code:
```c++
void CAndroidUtils::SetDecorViewBackgroundColor(const int color)
{
  CJNIWindow window = CXBMCApp::getWindow();

  if (window)
  {
    CJNIView view = window.getDecorView();
    if (view)
    {
      view.setBackgroundColor(color);
    }
  }
}
```

The app segfaults.

quietvoid commented:
All right, it works with `CJNIView` now.
Depends on https://github.com/xbmc/libandroidjni/pull/42

joseluismarti commented:
I was just about to try it and I see you've already found the cause :+1:

quietvoid commented on xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:
```diff
@@ -769,6 +771,13 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
+    {
+      CLog::Log(LOGDEBUG, "CDVDVideoCodecAndroidMediaCodec::Open Dolby Vision playback, setting "
+                          "background color to black");
+      CXBMCApp::Get().SetDecorViewBackgroundColor((signed int) 0xff000000);
```
It would probably cause a regression for #14865 to set it default.

fritsch replied:
Funny - when the past catches. I am honestly wondering if the change back then was a good change or if this transparency might have caused massive performance issues as stuff needs to alpha blend ...

Quite expensive that Sony workaround :-)

fritsch replied:
Maybe it would have been wise, if we only set it to Transparent whenever we play HDR content ...

fritsch replied:
What about: We revert the change from the past - and actually see if Sony catched up? :-) it's more than 4 years ago ... Would that at least work?

If Sony still did not catch up, we could use your libjniandroid change fix their HDR only ... without impacting entire GUI?

quietvoid replied:
I'll try reverting the old patch to see if it fixes the same issue this PR does, a bit later.

fritsch replied:
Jep, sadly. Having a "global workaround" for Sony TVs inside kodi ... is really something that is not great ... and especially if it enforces this again hacky workaround for other devices on top ... but this is what one gets if device specific hackarounds start :-(

Will check expoplayer this evening if they changed something recently in that regard.

quietvoid replied:
With some searching there's apparently a way to modify the root view (which I presume is what is set in activity_main.xml).

quietvoid replied:
We can reset the main VideoLayout color with https://github.com/xbmc/libandroidjni/pull/43
So the question is if we simply revert to a black background in the XML, or also add a workaround to change to transparent for HDR10 (and HLG?).

fritsch replied:
Let's revert the original "workaround". We have a saying in kodi: Don't add workarounds for specific devices ... the old "workaround" that impacts _every_ device should have never been merged ... even though the usual load screaming people were even louder that time.

I'd say:
Let's go for the original Revert, let's prepare the libjniandroid ... and if they come back, then yes - let's change the view exactly for them ... to not have everyone else suffer.

Sounds like a plan?

quietvoid replied:
Sure, I'll update the commit and PR info.
Thanks!

quietvoid committed:
[Android] Revert main layout background transparency to black

Reverts the workaround from 6f984677ed840f391d78307955a7df39f3016eb8

It causes issues on newer devices, and we can do better than
setting the main layout transparent globally in the XML.

If it causes regressions, see https://github.com/xbmc/libandroidjni/pull/43

fritsch reviewed (approved):
Fine with me. We should not workaround for older devices by changing requirements inside kodi, especially if they are additionally causing more CPU usage.

We can fix the outtage, if such other devices still exist when they come up.

fritsch commented:
Hi, sorry - that it took so long. Could you please rebase, so that the commit is on top? I want to try things out in master to see if things popup that we did not have in mind.

quietvoid commented:
Hello, no worries. I have rebased.

fritsch commented:
Thanks again! Here we go.

quietvoid commented:
Noting that with the new Amazon FireTV Stick 4K Max 2nd edition (2023), videos have grey bars in HDR10 now.
Will see if it gets fixed eventually, it affects the Amazon Prime app as well since they use a black background.

Dolby Vision still needs the black background for letterboxing to be black, so nothing changed for that.
